lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -196,28 +213,72 @@ def _get_device(self) -> str:
         else:
             return self.device
 
-    def infer(self, image: Image.Image, threshold: float) -> list[Element]:
-        inputs = self.processor(images=image, return_tensors="pt").to(self._get_device())
-        outputs = self.model(**inputs)
-        target_sizes = torch.tensor([image.size[::-1]])
-        results = self.processor.post_process_object_detection(outputs, target_sizes=target_sizes, threshold=threshold)[
-            0
-        ]
-        # need to wrap up the results in elements
-        elements = []
-        (w, h) = image.size
-        for score, label, box in zip(
-            results["scores"].cpu().detach().numpy(),
-            results["labels"].cpu().detach().numpy(),
-            results["boxes"].cpu().detach().numpy(),
-        ):
-            element = create_element(
-                type=self.labels[label],
-                bbox=BoundingBox(box[0] / w, box[1] / h, box[2] / w, box[3] / h).coordinates,
-                properties={"score": score},
-            )
-            elements.append(element)
-        return elements
+    def infer(
+        self, images: List[Image.Image], threshold: float, model_server_endpoint: str = ""
+    ) -> List[List[Element]]:
+        if model_server_endpoint:
+            endpoint = model_server_endpoint + self._model_name_or_path
+            metadata = {
+                "threshold": threshold,
+                "modes": [image.mode for image in images],
+                "sizes": [image.size for image in images],
+            }
+            metadata_string = json.dumps(metadata)
+            files = [("metadata", gzip.compress(metadata_string.encode("utf-8")))] + [
```

Comment:
I just notice this part, how this gzip performs considering the tradeoff of network load reduction and cpu compress/decompress overhead?